modules/caddyhttp/rewrite/rewrite.go

```diff
@@ -470,5 +478,45 @@ func changePath(req *http.Request, newVal func(pathOrRawPath string) string) {
 	}
 }
 
+// queryOps describes the operations to perform on a query like add, delete and replace fields
+type queryOps struct {
+	// Adds query parameters; does not replace any existing header fields.
+	Add []queryOpsArguments `json:"add,omitempty"`
+
+	// Sets query parameters; replaces existing header fields.
+	Set []queryOpsArguments `json:"set,omitempty"`
+
+	// Names of query parameters to delete
+	Delete []string `json:"delete,omitempty"`
+}
+
+func (q *queryOps) do(r *http.Request, repl *caddy.Replacer) {
+	query := r.URL.Query()
+
+	for _, addParam := range q.Add {
+		key := repl.ReplaceAll(addParam.Key, "")
```

Comment:
I think maybe if the key is empty it should do nothing. I don't think it's valid to have an empty query key? Need to check the HTTP spec to make sure but I'm pretty sure it's invalid.

I think it might also be invalid to have `=` in the key, but maybe Go encodes that? Can cover with a test.

Reply:
I just skipped the iteration if the key is empty. I also added a key with an `=` sign to the url in the test I wrote and it looks like go encodes that.